### PR TITLE
first prototype for the permissions service api

### PR DESCRIPTION
**Pull request by C0rby**
This is the proposed CS3 API extension to introduce a permissions service.
Currently this API is read-only but eventually should also allow to set or update permissions / assignments.

### CODE REVIEW

**Comment by refs** [comment]
@ishank011 could you review this one please?

File: cs3/permissions/v1beta1/permissions_api.proto [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+import "cs3/permissions/v1beta1/resources.proto";
+import "cs3/rpc/v1beta1/status.proto";
+import "cs3/storage/provider/v1beta1/resources.proto";
+
+// PermissionsAPI defines a service for permissions.
+service PermissionsAPI {
+  // CheckPermission defines a method to check permission/role.
+  rpc CheckPermission(CheckPermissionRequest) returns (CheckPermissionResponse);
+  // Could be added later on:
```
Comment (refs) [inline cs3/permissions/v1beta1/permissions_api.proto:39]: I think we can get rid of this comment here

**Comment by refs** [comment]
@C0rby rebase? 👼

**Comment by refs** [comment]
Reva implementation lives in https://github.com/cs3org/reva/pull/2341

**Comment by C0rby** [comment]
Ah, I messed something up. There are changes in this PR which are not from me. I'll fix that and then it should be good to go.

**Comment by C0rby** [comment]
I don't really know why this is here: https://github.com/cs3org/cs3apis/pull/161/files#diff-b04b38d4e36f7a7171aeb211bf933eaf36d41d9866ebbc3639f673f84dc350aeL5710-R5906.

Rebasing didn't remove it. I don't find any place where I am touching this in the proto files..

**Pull request merged**